### ISSUE TITLE
show titles in poster view "all media"

### ISSUE THREAD

**Issue by SonicPotions** [jellyfin/jellyfin-androidtv#654]
**Describe the feature you'd like**	
Currently when I view all media, only the posters/covers are shown, but no text below it with the title of the album/movie.
The title is only shown for the selected media. This is okayish if you have all covers, but especially for music where I'm missing covers there is no way to see what album it is when the album is not selected.

**Comment by Sveninndh** [comment]
Same issue for fotos or private home video libs.

**Comment by acordtag** [comment]
This request helps a lot visually when navigating on the "All items" view.

**Comment by oneseventhree** [comment]
Switching back to Emby because the posters view on AndriodTV app is much better due to having Title, Year, Resolution of video. Please implement! :)

**Comment by scottrobertson** [comment]
This would be really good. It's so hard to navigate movies/tv shows without titles.

**Comment by rpm5099** [comment]
Lots of requests for this in #1778, I think this issue is not being found because of language barrier. Highly desired feature, very difficult to identify shows in the android tv app.

**Comment by kiwichick888** [comment]
I also want this. Especially as the poster titles are displayed in the web app and Android mobile - it makes no sense to no have them displayed for Android TV (and any other platforms that don't have them). Better spacing between each title for Android TV is also needed.

Web:
<img width="480" alt="Image" src="https://github.com/user-attachments/assets/6cd87180-da9a-4fa8-84c6-2fa27500cbbd" />

Android mobile:
![Image](https://github.com/user-attachments/assets/0ebf0b48-aaed-4e1a-9b83-c7b70d7a5c35)

Android TV:
![Image](https://github.com/user-attachments/assets/6b0114d6-894c-401b-abdb-706e7e261f64)

**Comment by hard2carry** [comment]
I also think its hard to navigate as of now. The users eyes always have to bounce from the selection to the top for the title and its really tiring when browsing a library.
Showing the title underneath the poster / cover would improve the user experience by a lot. Its pretty much standard for most tv apps and also the android and web version of jellyfin.

**Comment by scottrobertson** [comment]
If I spent some time learning how to code for Android, would you accept a PR for this?

**Comment by nielsvanvelzen** [comment]
Yes pull requests properly implementing this would be accepted. Unfortunately the current code for the item grid is extremely messy and has a lot of hardcoded calculations that all need to be adapted and tested so it might not be trivial.

**Comment by GenericMaleUser** [comment]
@nielsvanvelzen Is this where the item grid code you are referring to is located?

https://github.com/jellyfin/jellyfin-androidtv/blob/master/app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java#L447

**Comment by nielsvanvelzen** [comment]
Correct, that entire file is the basis for the grid views.

**Comment by razor67rus** [comment]
The problem is in the **buildAdapter()** method.

https://github.com/jellyfin/jellyfin-androidtv/blob/master/app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java#L611

In the first line
`mCardPresenter = new CardPresenter(false, mImageType, mCardHeight);`
the first parameter is hardcoded as **false**

If you change it to **true**, it looks pretty good. I only tested it on the TV emulator in Android Studio.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/1826f213-cf55-4ecb-a99a-a423f85ceb24" />

**Comment by rpm5099** [comment]
> The problem is in the **buildAdapter()** method.
> 
> https://github.com/jellyfin/jellyfin-androidtv/blob/master/app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java#L611
> 
> In the first line `mCardPresenter = new CardPresenter(false, mImageType, mCardHeight);` the first parameter is hardcoded as **false**
> 
> If you change it to **true**, it looks pretty good. I only tested it on the TV emulator in Android Studio.
> 
> <img alt="Image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/90706758/502158925-1826f213-cf55-4ecb-a99a-a423f85ceb24.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA2MzQzMTgsIm5iZiI6MTc2MDYzNDAxOCwicGF0aCI6Ii85MDcwNjc1OC81MDIxNTg5MjUtMTgyNmYyMTMtY2Y1NS00ZWNiLWE5OWEtYTQyM2Y4NWNlYjI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE2VDE3MDAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzNjA3ZDJiZTA1ZWZmZDQxNDVkMWYxOWM5ZjgyMjc5M2I2YmYxNDcyNWZmZDg3ZmM0NmIyYWRlNjVhOWE3YTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PyY7f7MZYn7EkN_HVbezmOEa2Z30o2MdGhgLzafmCDw">

If this works as a stopgap until someone is able to find the time to clean up the libraries, anything is better than the current situation.  10 minutes with a room full of people waiting for you to find an unlabeled thumbnail is not fun.

**Comment by kiwichick888** [comment]
> If this works as a stopgap until someone is able to find the time to clean up the libraries, anything is better than the current situation.

Fantastic! This should be implemented as a workaround immediately.

**Comment by razor67rus** [comment]
@nielsvanvelzen can you please tell me if it is currently acceptable to implement the grid functionality on **Compose**? I know there were some focus issues. 
I am not an Android developer, but I am very concerned about this problem with the grid visualization, and I would really like to contribute to the development. Just because of this, I installed Android Studio to figure it out and try to rewrite it to **Compose**. 
It's like there aren't any critical focus issues in sight. Of course, there are some peculiarities due to the fact that the **Compose** component is wrapped in a Fragment, but this is all solvable.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/79d4e0f1-a3b2-4735-ac55-d66a5177842c" />

At the moment, about 70% of the grid functionality has been implemented on **Compose**. 
So, is it acceptable to use **Compose** at this point? And is it possible to use the **material** and **material3** libraries? They would simplify the development.

**Comment by nielsvanvelzen** [comment]
The focus issues with compose are no longer an issue. Only when wrapping the legacy UI components (specifically leanback views) but those will all be rewritten anyway. We cannot use the compose material libraries as those are not meant for televisions hence we're implementing our theme completely from scratch.

The browsing UI (including search results, the home screen, the smart screen and ofc. the item grids) is one of the things I've been working on since over a year ago but other priorities keep me from working on it so it's unfortunately not ready yet. I'd be open to pull requests that gradually move parts over to compose.

**Comment by khalob** [comment]
Adding the series title seems like low hanging fruit on the surface, but this ticket/issue is 5 years old. Is a PR targeting specifically that acceptable/warranted at this point?

**Comment by kiwichick888** [comment]
> this ticket/issue is 5 years old

The fact that this still hasn't been addressed is the reason I switched to using Wholphin as my Android TV client. It does display the titles and, having come to Jellyfin from Plex, the Wholphin interface has been designed to look like Plex.

**Comment by rpm5099** [comment]
> Adding the series title seems like low hanging fruit on the surface, but this ticket/issue is 5 years old. Is a PR targeting specifically that acceptable/warranted at this point?

Same.  It looks like that is the intended behavior.  It's understandable given how much volunteer work projects like Jellyfin are.  I wish I had the time to contribute myself, although I have little patience for the constraints of the Android ecosystem, especially with the impending doom of Android as we know it.  An official offloading of the client applications to a separate project may be worth considering.
